### PR TITLE
[WFLY-13959]: Add critical-analyzer configuration to WildFly Management

### PR DESCRIPTION
**Pull request by ehsavoie**
API.

### CODE REVIEW

File: messaging/WFLY-13959_critical_analyzer.adoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+=== Non-Requirements
+
+N/A
+
+== Implementation Plan
+
+* Add an optional BOOLEAN `critical-analyzer-enabled` attribute, to enable / disable the critical-analyzer, with a default value set to `true`.
+* Add a positive LONG `critical-analyzer-check-period` attribute, with a default value set to `0` in milliseconds. If the value is set to `0` then `critical-analyzer-timeout` /2 will be used instead.
+* Add a positive LONG `critical-analyzer-timeout` attribute, with a default value set to `120000` in milliseconds.
+* Add a STRING `critical-analyzer-policy` attribute with a value in the list `LOG`, `SHUTDOWN`, `HALT`, with a default value set to `LOG`.
```
Comment (honza-kasik) [inline messaging/WFLY-13959_critical_analyzer.adoc:62]: Similarly to these values - if critical analyzer is turned on by default now, these are values in use?

File: messaging/WFLY-13959_critical_analyzer.adoc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Those attributes will be grouped in the `critical-analyzer` attribute group for the https://wildscribe.github.io/WildFly/20.0/subsystem/messaging-activemq/server/[/subsystem=messaging-activemq/server resource].
+
+
+== Test Plan
+
+The tests will check that setting the attribute values and reloading will return the new values and that the server will start properly.
```
Comment (honza-kasik) [inline messaging/WFLY-13959_critical_analyzer.adoc:69]: Is there existing coverage in any of Artemis tests?

File: messaging/WFLY-13959_critical_analyzer.adoc [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+* https://issues.jboss.org/browse/WFLY-13959[WFLY-13959]
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
```
Comment (honza-kasik) [inline messaging/WFLY-13959_critical_analyzer.adoc:23]: That would be me `Jan Kasik <jkasik@redhat.com>`

**Comment by honza-kasik** [comment]
Thank you @ehsavoie ! Please see my comments.

File: messaging/WFLY-13959_critical_analyzer.adoc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Implementation Plan
+
+* Add an optional BOOLEAN `critical-analyzer-enabled` attribute, to enable / disable the critical-analyzer, with a default value set to `true`.
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging/WFLY-13959_critical_analyzer.adoc:59]: Yes

File: messaging/WFLY-13959_critical_analyzer.adoc (hunk 1 above)
Comment (ehsavoie) [reply messaging/WFLY-13959_critical_analyzer.adoc:62]: yes

File: messaging/WFLY-13959_critical_analyzer.adoc (hunk 2 above)
Comment (ehsavoie) [reply messaging/WFLY-13959_critical_analyzer.adoc:69]: Not that I'm aware of but this feature as been there for ages.

Comment (ehsavoie) [reply messaging/WFLY-13959_critical_analyzer.adoc:70]: Only unit tests in Artemis (upstream) testsuite, as far as I know.

Comment (honza-kasik) [reply messaging/WFLY-13959_critical_analyzer.adoc:70]: Thanks! This can be further discussed in test plan.

**Review by honza-kasik — APPROVED** [review]
Thanks, this looks good from QE point of view.

**Pull request merged**